db/db_test.cc

```diff
@@ -2587,15 +2585,23 @@ struct MTThread {
   bool multiget_batched;
 };
 
+// Based on Peter Dillinger's suggestion, the threads have been changed from
+// detached threads to joinable threads
 static void MTThreadBody(void* arg) {
   MTThread* t = reinterpret_cast<MTThread*>(arg);
   int id = t->id;
   DB* db = t->state->test->db_;
   int counter = 0;
+  auto start = std::chrono::high_resolution_clock::now();
```

Comment:
How about

```
std::shared_ptr<SystemClock> clock = SystemClock::Default();
auto end_micros = clock->NowMicros() + kTestSeconds * 1000000U;
...
while (clock->NowMicros() < end_micros) {
```

(And change `kTestSeconds` to unsigned)

Reply:
Done